models/login_source.go

```diff
@@ -384,6 +384,10 @@ func composeFullName(firstname, surname, username string) string {
 	}
 }
 
+var (
+	alphaDashDotPattern = regexp.MustCompile("[^\\d\\w-_\\.]")
```

Comment:
```suggestion
	alphaDashDotPattern = regexp.MustCompile("[^\\w-\\.]")
```
This could just be ```[^\\w-\\.]``` since \w is equal to ``` [a-zA-Z0-9_]```. Not a big deal though